### PR TITLE
Added methods to get names of sub-blocks in mae::Block

### PR DESCRIPTION
**Pull request by rachelnwalker**
This is helpful for reading other blocks in our MaeParserReader.

### CODE REVIEW

**Review by ricrogz — APPROVED** [review]
LGTM

File: MaeBlock.hpp [hunk 1]
```diff
@@ -8,6 +8,7 @@
 #include <string>
 #include <utility>
 
+#include "MaeConstants.hpp"
```
Comment (d-b-w) [inline MaeBlock.hpp:11]: wait - why does this change require the constants?

**Review by d-b-w — CHANGES_REQUESTED** [review]
lgtm after removing the extra include

File: MaeBlock.hpp (hunk 1 above)
Comment (rachelnwalker) [reply MaeBlock.hpp]: oh.. because I originally was not including atom/bond blocks but then decided to return the names of all blocks. Updated!

**Pull request merged**